metrics/core.py

```diff
@@ -94,27 +92,43 @@ def search(
         model = self.models[collection_name]
         embedding = self._single_img_infer(model, img)
 
-        search_result = self.qdrant_client.search(
+        search_result = qdrant_client.search(
             collection_name=collection_name,
             query_vector=embedding,
             query_filter=None,  # TODO: add filtering feature
             limit=limit,
         )
         return search_result
 
-    def _get_random_similar_images(
-        self, collection_name: str, k: int = 25
-    ) -> Tuple[Image.Image, Image.Image]:
+    def get_best_choice_for_uploaded_image(
+        self, base_img: Image.Image, collection_name: str, benchmark: int, k: int = 25
+    ) -> tuple[Image.Image, list[Image.Image]]:
         """
         Search for similar images of random image from given collection.
-        Returns tuple of images [anchor_image, grid image of k most similar images (the biggest cosine similarity)
-        # TODO: probably going to be removed as it is only helper method
+        Returns tuple of images [anchor_image, grid image of k most similar images (the biggest cosine similarity)]
         """
-        ds_path = METRIC_TYPES_DIR / collection_name
-        file = random.choice(list(ds_path.iterdir()))
-        img = Image.open(file)
-        results = self.search(img, collection_name, limit=k)
-        imgs = [RESIZE_TRANSFORM(Image.open(r.payload["file"])) for r in results]
-        grid = make_grid(imgs)
-        grid_img = torchvision.transforms.ToPILImage()(grid)
-        return img, grid_img
+        results = self.search(base_img, collection_name, limit=k)
+        results_bench = [r for r in results if round(r.score, 4) >= benchmark / 100]
```

Comment:
You probably dont need to round floats in comparision

Reply:
I have to if I want to be coherent with front-end, and there I want to show a percentage with 2 decimal places.

Reply:
This is for output images, not input parameters.